File: app/src/main/java/com/neome/java/api/nucleus/base/dto/EnvSignal.java
// neome.ai API - do not change
//

package com.neome.java.api.nucleus.base.dto;

import com.neome.java.api.meta.base.Types.RequestId;
import com.neome.java.api.meta.base.Types.ServiceName;
import com.neome.java.api.meta.base.sig.ISig;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class EnvSignal<S extends ISig>
{
  @Nullable
  public Boolean cookieRememberMe;

  @Nullable
  public String cookieValue;

  @Nullable
  public EnvError error;

  @Nullable
  public RequestId requestId;

  @Nullable
  public String serverName;

  @Nullable
  public Long serverTime;

  @Nullable
  public ServiceName serviceName;

  @Nullable
  public S sig;

  @Nullable
  public String sigName;
}

File: app/src/main/java/com/neome/api/nucleus/base/dto/EnvSignal.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.nucleus.base.dto

import com.neome.api.nucleus.base.dto.EnvError
import com.neome.api.meta.base.sig.ISig
import com.neome.api.meta.base.Types.RequestId
import com.neome.api.meta.base.Types.ServiceName

interface EnvSignal<S : ISig>
{
  val cookieRememberMe: Boolean?
  val cookieValue: String?
  val error: EnvError?
  val requestId: RequestId?
  val serverName: String?
  val serverTime: Long?
  val serviceName: ServiceName?
  val sig: S?
  val sigName: String?
}